### PR TITLE
[WIP] (docs) Dart - dev guide/user input example

### PR DESCRIPTION
**Pull request by tomplusplus**
@kwalrath @kevmoo Please review. Should be easy. It's a slimmer version of the example for Template Syntax.
Thanks

### CODE REVIEW

File: public/docs/_examples/user-input/dart/lib/app_component.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      ClickMeComponent,
+      KeyUpComponent,
+      KeyUpComponentV2,
+      KeyUpComponentV3,
+      KeyUpComponentV4,
+      LoopBackComponent
+    ])
+class AppComponent {
+  onClickMe(event){
+    var evtMsg = event != null ? ' Event target class is '+ event.target.className  : '';
```
Comment (kevmoo) [inline public/docs/_examples/user-input/dart/lib/app_component.dart]: use String interpolation here

File: public/docs/_examples/user-input/dart/lib/app_component.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    window.alert('Click me.'+evtMsg);
```
Comment (kevmoo) [inline public/docs/_examples/user-input/dart/lib/app_component.dart]: and here – always basically.

File: public/docs/_examples/user-input/dart/lib/my-app.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+<h4>keyup loop-back component</h4>
+<loop-back></loop-back>
+
+<key-up></key-up>
+
+<key-up2></key-up2>
+
+<key-up3></key-up3>
+
+<key-up4></key-up4>
```
Comment (kevmoo) [inline public/docs/_examples/user-input/dart/lib/my-app.html]: include a newline at the end of file

File: public/docs/_examples/user-input/dart/web/index.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    <title>User Input</title>
+    <script async src="main.dart" type="application/dart"></script>
+    <script async src="packages/browser/dart.js"></script>
+</head>
+<body>
+    <my-app>Loading...</my-app>
+    <hr>
+    <little-tour>Loading...</little-tour>
+</body>
+</html>
```
Comment (kevmoo) [inline public/docs/_examples/user-input/dart/web/index.html]: ditto

File: public/docs/_examples/user-input/dart/lib/key_up_components.dart [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+@Component(selector: 'key-up')
+@View(
+    template: '''<h4>Give me some keys!</h4>
+    <div><input (keyup)="onKey(\$event)"><div>
+    <div>{{values}}</div>''')
+class KeyUpComponent {
+  String values = '';
+  onKey(event) {
+    values += event.target.value + ' | ';
```
Comment (kevmoo) [inline public/docs/_examples/user-input/dart/lib/key_up_components.dart]: Is this what the TS script version does? string concat hurts my head...

Comment (tomplusplus) [inline public/docs/_examples/user-input/dart/lib/key_up_components.dart]: I agree - I'll update it.

File: public/docs/_examples/user-input/dart/lib/little_tour.dart [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      (keyup.enter)="addHero(newHero)"
+      (blur)="addHero(newHero)">
+    <button (click)=addHero(newHero)>Add</button>
+    <ul><li *ng-for="#hero of heroes">{{hero}}</li></ul>''',
+    directives: const [CORE_DIRECTIVES, FORM_DIRECTIVES])
+class LittleTourComponent {
+  List<String> heroes = ['Windstorm', 'Bombasto', 'Magneta', 'Tornado'];
+
+  void addHero(InputElement newHero) {
+    if (newHero.value != null && newHero.value.length > 0) {
```
Comment (kwalrath) [inline public/docs/_examples/user-input/dart/lib/little_tour.dart]: Could you use `newHero?.value?.length > 0` here?

File: public/docs/_examples/user-input/dart/web/index.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<!-- #docregion -->
+<!DOCTYPE html>
+<html>
+<head>
+    <title>User Input</title>
```
Comment (kwalrath) [inline public/docs/_examples/user-input/dart/web/index.html]: 2-space indent

**Comment by tomplusplus** [comment]
@kevmoo @kwalrath Made the updates suggested. And changed to mach TS. please review.

Tom

**Comment by kwalrath** [comment]
Other than the two small things I pointed out, this LGTM.

**Comment by kwalrath** [comment]
I've started working on this chapter, so I'm going to merge this and make the last couple of changes myself.

**Pull request closed without merging**